Se agregaron la dependencias de Spring Security, y de JWT token.

ivanochoa229 opened the pull request:
Se implementaron las clases e implementaciones necesarias para que funcione la cadena de Filtros de Spring Security. Se agrego un archivo .sql para agregar los campos necesarios para las columnas del user que solicita spring.

ivanochoa229 committed:
Se agregaron la dependencias de Spring Security, y de JWT token.
Se implementaron las clases e implementaciones necesarias para que funcione la cadena de Filtros de Spring Security.
Se agrego un archivo .sql para agregar los campos necesarios para las columnas del user que solicita spring.

CabaleiroNicolas commented on src/main/java/com/horapp/persistence/entity/User.java:
De la propiedades que porporciona el UserDetails yo creo que deberiamos usar solamente la de enabled (que esta bueno que lo cambiaste por deleted) y tambien puede ser la de Locked, pero la de las credenciales creo que no

CabaleiroNicolas commented on src/main/java/com/horapp/presentation/controller/UserController.java:
El metodo para crear un usuario lo debe manejar el servicio de usuario, no el de autenticacion

CabaleiroNicolas commented on src/main/java/com/horapp/persistence/entity/User.java:
De la propiedades que porporciona el UserDetails yo creo que deberiamos usar solamente la de enabled (que esta bueno que lo cambiaste por deleted) y tambien puede ser la de Locked, pero la de las credenciales creo que no.

Tambien, no usaremos Setters solo getters, de manera que para crear una instancia solo lo haremosmediante constructores

CabaleiroNicolas commented on src/main/java/com/horapp/presentation/controller/auth/AuthenticationController.java:
Para que es el endpoint de valida token?

ivanochoa229 replied:
Ese endpoint  no es necesario, lo copié del proyecto que me guiaba y me olvidé de eliminarlo.

CabaleiroNicolas commented on src/main/java/com/horapp/presentation/dto/response/UserResponseDTO.java:
Ver: si hacemos que el usaurio tenga que confirmar el mail, no hay que devolver el jwt de inicio de sesion en el registro

CabaleiroNicolas commented on src/main/java/com/horapp/service/auth/AuthenticationService.java:
Aqui no va el metodo para hacer el Save (iria en UserService), y tampoco el metodo de validar el token (iria en JwtService)

CabaleiroNicolas commented on src/main/java/com/horapp/service/impl/UserServiceImpl.java:
No hay que validar con el if si estan vacios los campos en el metodo findAll(), hay que validar con las anotaciones de Java Validation en el DTO de request, tampoco devolver un nullPointerExcprion porque esta excepcion no conviene manejarla

CabaleiroNicolas commented on src/main/resources/db/migration/V2__USER_UPDATE.sql:
trabajar en la migracion INITIAL, no hay problema, ya que trata de la misma configuracion, solo hay que vaciar la tabla que genera flyway en local :)

ivanochoa229 committed:
Se agregaron los cambios al archivo principal de migraciones.
Se elimino el campo credentials_non_experied de User
Se elimino el endpoint para validar el token,
Al momento de registrar un user, se hace desde el UserService, y ahora devuelve un mensaje de exito.
Se agregaron las validaciones con Validate de jakarta, junto con su dependencia en el pom.xml
Se creo un nuevo dto para el manejo de las excepciones por validacion de campos
Se agregaron fetch type eager en las entidades que lo requerian
